src/audio/codec_adapter/codec/cadence.c

```diff
@@ -667,31 +667,11 @@ cadence_codec_set_configuration(struct processing_module *mod, uint32_t config_i
 				const uint8_t *fragment, size_t fragment_size, uint8_t *response,
 				size_t response_size)
 {
-	struct module_data *md = &mod->priv;
-	struct comp_dev *dev = mod->dev;
-	int ret;
-
-	ret = module_set_configuration(mod, config_id, pos, data_offset_size, fragment,
-				       fragment_size, response, response_size);
-	if (ret < 0)
-		return ret;
-
-	/* return if more fragments are expected or if the module is not prepared */
-	if ((pos != MODULE_CFG_FRAGMENT_LAST && pos != MODULE_CFG_FRAGMENT_SINGLE) ||
-	    md->state < MODULE_INITIALIZED)
-		return 0;
 
-	/* whole configuration received, apply it now */
-	ret = cadence_codec_apply_config(dev);
-	if (ret) {
-		comp_err(dev, "cadence_codec_set_configuration(): error %x: runtime config apply failed",
-			 ret);
-		return ret;
-	}
-
-	comp_dbg(dev, "cadence_codec_set_configuration(): config applied");
+	comp_dbg(mod->dev, "cadence_codec_set_configuration()");
 
-	return 0;
+	return module_set_configuration(mod, config_id, pos, data_offset_size, fragment,
```

Comment:
would it better to remove cadence_codec_set_configuration() and set ..set_configuration = model_set_configuration directly?

Reply:
That would be an option too. I will add it in a separate patch to be more obvious.